lib/tablepool.lua

```diff
@@ -8,6 +8,18 @@ local pools = newtab(0, 4)
 
 
 function _M.fetch(tag, narr, nrec)
+    if not tag then
+        error("tag empty", 2)
+    end
+
+    if not narr then
+        error("narr empty", 2)
+    end
+
+    if not nrec then
+        error("nrec empty", 2)
+    end
```

Comment:
This incurs extra overhead (remember `table.move()` is obviously slower than a hand-crafted Lua loop?)